clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
     public Cluster(Collection<Node> nodes,
                    Collection<PartitionInfo> partitions,
                    Set<String> unauthorizedTopics) {
-        this(false, nodes, partitions, unauthorizedTopics);
+        this(false, nodes, partitions, unauthorizedTopics, Collections.<String>emptySet());
```

Comment:
How would there be a change in semantics in `MockProducer` given that it doesn't care about internal topics? The only place we currently use internal topics is the `ConsumerCoordinator`.